pipe-cli/src/utilities/storage/s3.py

```diff
@@ -300,6 +303,30 @@ def load_delete_marker(self, bucket, path, client):
                     return item
         raise RuntimeError('Latest file version is not deleted. Please specify "--version" parameter.')
 
+    def restore_folder(self, bucket, client, exclude, include):
+        delimiter = S3BucketOperations.S3_PATH_SEPARATOR
+        path = self.bucket.path
+        prefix = StorageOperations.get_prefix(path)
+
+        operation_parameters = {
+            'Bucket': bucket,
+        }
+        if path:
+            operation_parameters['Prefix'] = prefix
+        paginator = client.get_paginator('list_object_versions')
+        pages = paginator.paginate(**operation_parameters)
+        restore_us = dict(Objects=[])
+        for page in pages:
+            S3BucketOperations.process_listing(page, 'DeleteMarkers', restore_us, delimiter, exclude, include, prefix,
+                                               versions=True)
+            # flush once aws limit reached
+            if len(restore_us['Objects']) >= 1000:
```

Comment:
It seems, that this code will fail if `len(restore_us['Objects']) > 1000`

Reply:
fixed for deletion too